crates/mysten-network/src/multiaddr.rs

```diff
@@ -101,6 +101,17 @@ impl Multiaddr {
         Ok(SocketAddr::new(ip, tcp_port))
     }
 
+    // Returns Ok(true) if the Multiaddr is a valid TCP address.
+    pub fn is_tcp_addr(&self) -> Result<bool> {
+        let mut iter = self.iter();
+        iter.next(); // Skip the ip/dns part
+        Ok(matches!(
```

Comment:
Is an address tcp if it contains both tcp and udp address?
Also curious, is being tcp the only constraint here? I wonder if we need to go as far as trying to call the same tcp parse function to make sure.

Reply:
an eligible address can't contain both tcp and udp addresses at the same time.

> Also curious, is being tcp the only constraint here? 
I think so. I don't know if we support anything other than ipv4/ipv6/dns though.

> call the same tcp parse function
if you are referring to the existing tcp parse function, that does not work because it expects the immediate next `protocol` to be `tcp`. Other than that it does the same thing as `fn is_tcp_addr`

Reply:
What do you mean by "that does not work"?

Reply:
do you mean `fn parse_tcp` https://github.com/MystenLabs/sui/blob/main/crates/mysten-network/src/multiaddr.rs#L249? 
Note it takes an iterator and expects the `next()` value to be `tcp`. Hence, if you have `"/ip4/127.0.0.1/tcp/"` it errs. You need to give sth like "/tcp/"

Reply:
I may not be following what you mean, but we don't have a generic `the same tcp parse function` yet